blog-samples/CSharp/FacebookHandover/Primary/Bots/PrimaryBot.cs

```diff
@@ -0,0 +1,213 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+//
+// Generated with Bot Builder V4 SDK Template for Visual Studio EchoBot v4.3.0
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Dialogs.Choices;
+using Microsoft.Bot.Schema;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json.Linq;
+using FacebookModel;
+using System;
+using Newtonsoft.Json;
+using Microsoft.Bot.Connector;
+
+namespace Primary.Bots
+{
+    /// <summary>
+    /// This is based on Eric Dahlvang's FacebookEventsBotExpanded: https://github.com/EricDahlvang/FacebookEventsBotExpanded
+    /// </summary>
+    public class PrimaryBot : ActivityHandler
+    {
+        private const string OPTION_PASS_PAGE_INBOX = "Pass to page inbox";
+        private const string OPTION_PASS_SECONDARY_BOT = "Pass to secondary";
+        private const string OPTION_REQUEST_THREAD_CONTROL = "Receive request";
+        private const string OPTION_REQUEST_THREAD_CONTROL_NICELY = "Receive nice request";
+        private const string OPTION_TAKE_THREAD_CONTROL = "Have control taken";
+        private const string PAGE_INBOX_ID = "263902037430900";
+
+        private static readonly string[] _options = new[] { OPTION_PASS_PAGE_INBOX, OPTION_PASS_SECONDARY_BOT, OPTION_REQUEST_THREAD_CONTROL, OPTION_REQUEST_THREAD_CONTROL_NICELY };
```

Comment:
This could be a static list of Choice (so it is only created once):
        private static readonly Choice[] _options = new[] { new Choice(OPTION_PASS_PAGE_INBOX),
                                                            new Choice(OPTION_PASS_SECONDARY_BOT),
                                                            new Choice(OPTION_REQUEST_THREAD_CONTROL),
                                                            new Choice(OPTION_REQUEST_THREAD_CONTROL_NICELY) };